Sources/MongoSwift/ReadConcern.swift

```diff
@@ -86,13 +115,19 @@ public class ReadConcern: Codable {
         try container.encodeIfPresent(self.level, forKey: .level)
     }
 
-    /// Cleans up internal state.
-    deinit {
-        guard let readConcern = self._readConcern else {
-            return
+    /**
+     * Creates a new `ReadConcern` with the provided options and passes it to the provided closure.
+     * The read concern is only valid within the body of the closure and will be ended after the body completes.
+     */
+    public func withMongocReadConcern<T>(_ body: (OpaquePointer) throws -> T) throws -> T {
```

Comment:
This method is only used by us internally and isn't part of the public API, so it can have an `internal` access level control modifier. Also, since the outer portion of this method doesn't throw, it can be marked as `rethrows`, indicating it will only throw if the closure does. This will allow us to omit `try` in cases in which the closure we pass in doesn't throw. See the "rethrowing functions and methods" portion of https://docs.swift.org/swift-book/ReferenceManual/Declarations.html for more info.